src/Security/Security.php

```diff
@@ -423,7 +423,10 @@ public static function permissionFailure($controller = null, $messageSet = null)
         list($messageText, $messageCast) = $parseMessage($message);
         static::singleton()->setSessionMessage($messageText, ValidationResult::TYPE_WARNING, $messageCast);
 
-        $controller->getRequest()->getSession()->set("BackURL", $_SERVER['REQUEST_URI']);
+        $session = $controller->getRequest()->getSession();
```

Comment:
I don’t think this will fix #9117 😞just calling `getSession()` [throws an exception](https://github.com/silverstripe/silverstripe-framework/blob/4/src/Control/HTTPRequest.php#L890-L892). Perhaps the simplest fix would be to wrap it in a try/catch?

Reply:
Good point 🤦‍♂